src/client/connect/mod.rs

```diff
@@ -115,6 +118,114 @@ pub struct Connected {
     pub(super) extra: Option<Extra>,
 }
 
+/// [`CaptureConnection`] allows callers to capture [`Connected`] information
```

Comment:
Does this need to live in this file or can we move it out?

Reply:
can definitely be moved—any preference for final location?